lib/src/command/update_flutter_sdk_command.dart

```diff
@@ -40,53 +42,59 @@ class UpdateFlutterSdkCommand extends MoronepoCommand<Null> {
-        print("Running flutter version ${forcedVersion} for ${rootProject.name} project");
-        await _enforceVersion(forcedVersion, rootProject.path);
+        await _fetchFlutterSdk(flutterSdkPath);
+        final availableTags = await _fetchAvailableFlutterVersions(flutterSdkPath);
+        final forcedTag = _determineBestVersion(versionConstraint, availableTags);
+        print("Enforcing version ${forcedTag.version} for ${rootProject.name} project");
+        await _enforceVersion(forcedTag, flutterSdkPath);
+        await _runFlutterDoctor(flutterSdkPath);
       }
     }
   }
 
   Future<Version> _fetchFlutterVersion(String path) async {
     final processOutput = await _processStarter.start("flutter", ["--version"], path);
-    final version = RegExp(r"Flutter ([^\s]+)").firstMatch(processOutput.output).group(1);
+    final version =
+        RegExp(r"Flutter ([^\.]+\.[^\.]+\.[^\s]*)").firstMatch(processOutput.output).group(1);
     return Version.parse(version);
   }
 
-  Future<void> _fetchFlutterSdk() async {
-    final flutterSdkPath = _flutterFinder.findFlutter();
+  Future<void> _fetchFlutterSdk(String flutterSdkPath) async {
     return _processStarter.start("git", ["fetch"], flutterSdkPath);
   }
 
-  Future<List<Version>> _fetchAvailableFlutterVersions(String path) async {
-    final processOutput = await _processStarter.start("flutter", ["version"], path);
+  Future<List<Tag>> _fetchAvailableFlutterVersions(String path) async {
+    final processOutput = await _processStarter.start("git", ["tag", "-l", "*.*.*"], path);
     final versionStrings = processOutput.output.split("\n");
     return versionStrings
         .where((it) => it.isNotEmpty)
-        .map((it) => Version.parse(it.substring(1)))
+        .map((it) => Tag(it))
         .toList();
   }
 
-  Version _determineBestVersion(VersionConstraint constraint, List<Version> availableVersions) {
-    final versionCandidates = availableVersions.where((it) => constraint.allows(it)).toList();
-    versionCandidates.sort((first, second) => Version.antiprioritize(first, second));
-    return versionCandidates.first ?? Version.none;
+  Tag _determineBestVersion(VersionConstraint constraint, List<Tag> availableTags) {
+    final versionCandidates = availableTags.where((it) => constraint.allows(it.version)).toList();
+    versionCandidates.sort((first, second) => Version.antiprioritize(first.version, second.version));
+    return versionCandidates.isNotEmpty ? versionCandidates.first : throw TagNotFoundException(constraint);
   }
 
-  Future<void> _enforceVersion(Version forcedVersion, String path) async {
+  Future<void> _enforceVersion(Tag forcedTag, String path) async {
     await _processStarter.start(
-      "flutter",
+      "git",
       [
-        "version",
-        forcedVersion.toString(),
+        "checkout",
+        forcedTag.value,
       ],
       path,
     );
   }
+
+  void _runFlutterDoctor(String path) async {
```

Comment:
Flutter doctor to check if something is needed after checkout?

Reply:
I checked that I should be running `flutter precache` instead